source/4.0/learn/introduction/active-record.html.md

```diff
@@ -1,82 +1,545 @@
+making those validations so vague as to allow for different data configurations
+rendering them effectively meaningless forcing the need to test the model every
+time before use.
+
+Since it's so easy with ROM to load only pertinent data, we focus on allowing
+the user to map data into their desired format. Then using context aware
+validations at the system boundaries, developers can trust the data is correct
+and utilize it without fear.
+
+### Query with Complex Conditions
+
+<h4 class="text-center">Active Record</h4>
+```ruby
+User.where("admin IS ? OR moderator IS ?", true, true)
+```
+
+
+<h4 class="text-center">ROM</h4>
+```ruby
+users_relation.where { admin.is(true).or(moderator.is(true)) }
+```
+
+For several SQL keywords, such as `select` & `where`, ROM provides a DSL for
+blocks. The benefit is the ability to use any SQL functions supported by your
+database. 
+
+## Associations
+
+Similar to ActiveRecord, ROM uses associations as a means of describing the
+interconnections between data. 
+
+<!--
+  NOTE: Expand on this section with examples on how associations work
+        and how to configure them properly.
+-->
+
+### Join Query
+
+<h4 class="text-center">Active Record</h4>
+```ruby
+Article.joins(:users)
+```
+
+<h4 class="text-center">ROM</h4>
+```ruby
+articles_relation.join(:users)
+```
+
+Obviously the join interface for both frameworks can support different configurations
+to handle different types of joins, however this example illustrates that other than
+a minor name change, in the majority of use-cases they will act the same.
+
+## Persistence
+
+### Creating Simple Objects
+
+<h4 class="text-center">Active Record</h4>
+```ruby
+User.create(name: "Jane")
+#> #<User id: 1, name: "Jane">
+```
+
+
+<h4 class="text-center">ROM</h4>
+```ruby
+users_relation
+  .changeset(:create, name: "Jane")
+  .commit
+#> #<ROM::Struct::User id=1 name="Jane">
+```
+
+Changesets are an abstraction created over commands which are what actually
+manipulate stored records. They are preferred over commands due to additional
+functionality they provide. 
+
+
+### Updating Simple Objects
+
+<h4 class="text-center">Active Record</h4>
+```ruby
+user = User.find_by(name: "Jane")
+user.update(name: "Jane Doe")
+
+#> #<User id=1 name="Jane Doe">
+```
+
+
+<h4 class="text-center">ROM</h4>
+```ruby
+users_relation
+  .where(name: "Jane")
+  .changeset(:update, name: "Jane Doe")
+  .commit
+
+#> #<ROM::Struct::User id=1 name="Jane Doe">
+```
+
+It should be noted that updating a record in ActiveRecord generally requires 
+that record to first be loaded then updated then committed. We view this as
+a bad practice as it leads to more round trips from the database and entities
+that are initialized in an invalid state. If a developer is sufficiently
+validating data at the boundaries of the application then updating or creating
+a record without loading it should be no problem and in fact preferable.
+
+### Create Nested Objects
```

Comment:
Let's drop this section for now. We can add it once rom 5.0 is done, where nested changesets will work too.

Reply:
I've just commented out this section but left it in place for when it gets updated.